pkg/gardenlet/controller/shoot/shoot/reconciler_migrate.go

```diff
@@ -98,6 +99,9 @@ func (r *Reconciler) runMigrateShootFlow(ctx context.Context, o *operation.Opera
 			return nil
 		}),
 		errorsutils.ToExecute("Retrieve the infrastructure resource", func() error {
+			if isWorkerless {
+				return nil
+			}
 			obj, err := botanist.Shoot.Components.Extensions.Infrastructure.Get(ctx)
 			if err != nil {
 				if apierrors.IsNotFound(err) {
```

Comment:
check "Check whether control plane restoration is needed" for nil infrastructure case